src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -190,6 +197,33 @@ protected void checkConnection() throws SQLException {
         }
     }
 
+    private MongoClusterType determineClusterType() {
+        BsonDocument buildInfoCmd = new BsonDocument();
+        buildInfoCmd.put("buildInfo", new BsonInt32(1));
+
+        // The { buildInfo: 1 } command returns information that indicates
+        // the type of the cluster.
+        BuildInfo buildInfoRes =
+                mongoClient
+                        .getDatabase("admin")
+                        .withCodecRegistry(MongoDriver.registry)
+                        .runCommand(buildInfoCmd, BuildInfo.class);
+
+        // If the "dataLake" field is present, it must be an ADF cluster.
+        if (buildInfoRes.dataLake != null) {
+            return MongoClusterType.AtlasDataFederation;
+        } else if (buildInfoRes.modules != null) {
+            // Otherwise, if "modules" is present and contains "enterprise",
+            // this must be an Enterprise cluster.
+            if (buildInfoRes.modules.contains("enterprise")) {
+                return MongoClusterType.Enterprise;
+            }
+        }
+
+        // Otherwise, this is a Community cluster.
+        return MongoClusterType.Community;
```

Comment:
I see that the `buildInfo` command returns ` "ok" : <num>`.  Should we check that "ok" was `1`?  If this is something we should check for, that ok was 1, then maybe we would need an `unknown` cluster type when it can't be determined.  That way we wouldn't be defaulting to Community if we could not in fact determine which type it was.